core/audits/bf-cache.js

```diff
@@ -0,0 +1,141 @@
+   */
+  displayValue: `{itemCount, plural,
+    =1 {1 actionable failure reason}
+    other {# actionable failure reasons}
+    }`,
+};
+/* eslint-enable max-len */
+
+const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
+
+class BFCache extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'bf-cache',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      supportedModes: ['navigation', 'timespan'],
+      requiredArtifacts: ['BFCacheErrors'],
+    };
+  }
+
+  /**
+   * @param {LH.Crdp.Page.BackForwardCacheNotRestoredReason} reason
+   */
+  static getDescriptionForReason(reason) {
+    const matchingString = NotRestoredReasonDescription[reason];
+
+    if (matchingString === undefined) {
+      return reason;
+    }
+
+    return matchingString.name;
+  }
+
+  /**
+   * @param {LH.Artifacts.BFCacheErrorMap} errors
+   * @param {LH.IcuMessage | string} label
+   * @return {LH.Audit.Details.Table}
+   */
+  static makeTableForFailureType(errors, label) {
+    /** @type {LH.Audit.Details.TableItem[]} */
+    const results = [];
+
+    // https://github.com/Microsoft/TypeScript/issues/12870
+    const reasons = /** @type {LH.Crdp.Page.BackForwardCacheNotRestoredReason[]} */
+      (Object.keys(errors));
+
+    for (const reason of reasons) {
+      const frameUrls = errors[reason] || [];
+      results.push({
+        reason: this.getDescriptionForReason(reason),
+        subItems: {
+          type: 'subitems',
+          items: frameUrls.map(frameUrl => ({frameUrl})),
+        },
+      });
+    }
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      /* eslint-disable max-len */
+      {key: 'reason', valueType: 'text', subItemsHeading: {key: 'frameUrl', valueType: 'url'}, label},
+      /* eslint-enable max-len */
+    ];
+
+    return Audit.makeTableDetails(headings, results);
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts) {
+    const {PageSupportNeeded, SupportPending, Circumstantial} = artifacts.BFCacheErrors;
+
+    const actionableTable =
+      this.makeTableForFailureType(PageSupportNeeded, str_(UIStrings.actionableColumn));
+    const notActionableTable =
+      this.makeTableForFailureType(Circumstantial, str_(UIStrings.notActionableColumn));
+    const supportPendingTable =
+      this.makeTableForFailureType(SupportPending, str_(UIStrings.supportPendingColumn));
+
+    const items = [
+      actionableTable,
+      notActionableTable,
+      supportPendingTable,
+    ];
+
+    if (actionableTable.items.length === 0) {
+      return {
+        score: 1,
+        details: {
+          type: 'list',
+          items,
+        },
+      };
+    }
+
+    return {
+      score: 0,
```

Comment:
may want to mark n/a if `supportPendingTable` or `notActionableTable` is >0

Reply:
If there are still actionable failures I think it's worth surfacing them